frontend/src/main/components/Courses/CoursesForm.js

```diff
@@ -0,0 +1,156 @@
+    );
+    // Stryker restore all
+
+    const navigate = useNavigate();
+
+    const testIdPrefix = "CoursesForm";
+
+    // For explanation, see: https://stackoverflow.com/questions/3143070/javascript-regex-iso-datetime
+    // Note that even this complex regex may still need some tweaks
+
+    // Stryker disable next-line Regex
+    const isodate_regex = /(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+)|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d)|(\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d)/i;
+    
+
+    return (
+        <Form onSubmit={handleSubmit(submitAction)}>
+
+            {initialContents && (
+                <Form.Group className="mb-3" >
+                    <Form.Label htmlFor="id">Id</Form.Label>
+                    <Form.Control
+                        data-testid={testIdPrefix + "-id"}
+                        id="id"
+                        type="text"
+                        {...register("id")}
+                        value={initialContents.id}
+                        disabled
+                    />
+                </Form.Group>
+            )}
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="name">Name</Form.Label>
+                <Form.Control
+                    data-testid={testIdPrefix + "-name"}
+                    id="name"
+                    type="text"
+                    isInvalid={Boolean(errors.name)}
+                    {...register("name", {
+                        required: "Name is required.",
+                    })}
+                />
+                <Form.Control.Feedback type="invalid">
+                    {errors.name?.message}
+                </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="school">School</Form.Label>
+                <Form.Control
+                    data-testid={testIdPrefix + "-school"}
+                    id="school"
+                    type="text"
+                    isInvalid={Boolean(errors.school)}
+                    {...register("school", {
+                        required: "School is required."
+                    })}
+                />
+                <Form.Control.Feedback type="invalid">
+                    {errors.school?.message}
+                </Form.Control.Feedback>
+            </Form.Group>
+
+            <Form.Group className="mb-3" >
+                <Form.Label htmlFor="term">Term</Form.Label>
+                <Form.Control
+                    data-testid={testIdPrefix + "-term"}
+                    id="term"
+                    type="text"
+                    isInvalid={Boolean(errors.term)}
+                    {...register("term", {
+                        required: "Term is required."
+                    })}
+                />
+                <Form.Control.Feedback type="invalid">
+                    {errors.term?.message}
+                </Form.Control.Feedback>
+            </Form.Group>
+
```

Comment:
Could we use a `<Row><Col>...</Col></Row>` layout here so that these are side by side instead of on top of each other?

Instead of this:

<img width="1268" alt="image" src="https://github.com/ucsb-cs156-f23/proj-organic-f23-6pm-1/assets/1119017/de2cd5db-7936-4616-aa43-5947845db13c">

The fields are side by side like this:

<img width="959" alt="image" src="https://github.com/ucsb-cs156-f23/proj-organic-f23-6pm-1/assets/1119017/f3bf9537-fdbb-4441-94ad-c006521bbe08">


You can find an example of code to do this in this form:

https://github.com/ucsb-cs156/proj-happycows/blob/main/frontend/src/main/components/Commons/CommonsForm.js

And a storybook for that here to see what it looks like:

https://ucsb-cs156.github.io/proj-happycows/storybook/?path=/story/components-commons-commonsform--uncontrolled